resolwe/flow/managers/workload_connectors/kubernetes.py

```diff
@@ -499,17 +511,37 @@ def _image_mapper(self, image_name: str, mapper: Dict[str, str]) -> str:
                 return f"{mapper[key]}{image_name[len(key):]}"
         return image_name
 
+    @retry(
+        logger=logger,
+        max_retries=5,
+        retry_exceptions=(kubernetes.config.config_exception.ConfigException,),
+    )
+    def _load_kubernetes_config(self):
+        """Load the kubernetes configuration.
+
+        :raises kubernetes.config.config_exception.ConfigException: when the
+            configuration could not be read.
+        """
+        (
+            kubernetes.config.load_incluster_config()
+            if self._config_location == ConfigLocation.INCLUSTER
+            else kubernetes.config.load_kube_config()
+        )
+
     def start(self, data: Data, listener_connection: Tuple[str, str, str]):
         """Start process execution.
 
         Construct kubernetes job description and pass it to the kubernetes.
         """
+
         # Create kubernetes API every time otherwise it will time out
         # eventually and raise API exception.
+
         try:
-            kubernetes.config.load_kube_config()
+            self._load_kubernetes_config()
         except kubernetes.config.config_exception.ConfigException:
-            kubernetes.config.load_incluster_config()
+            logger.exception("Could not load the kubernetes configuration.")
+            raise
```

Comment:
Will this re-raise the same exception?

Reply:
Yes. For the sake of maintenance I have make the call explicit.